lib/src/modules/task.cc

```diff
@@ -128,11 +146,160 @@ static TaskCommand getTaskCommand(const fs::path &task_executable)
     return TaskCommand { task_executable.string(), { } };
 }
 
+// Converts boost filesystem errors to a task_error object.
+static Module::ProcessingError toModuleProcessingError(const fs::filesystem_error& e) {
+    auto err_code = e.code();
+    if (err_code == boost_error::no_such_file_or_directory) {
+        return Module::ProcessingError(lth_loc::format("No such file or directory: {1}", e.path1()));
+    } else if (err_code ==  boost_error::file_exists) {
+        return Module::ProcessingError(lth_loc::format("A file matching the name of the provided sha already exists"));
+    } else {
+        return Module::ProcessingError(e.what());
+    }
+}
+
+static const fs::perms NIX_TASK_FILE_PERMS = NIX_DIR_PERMS;
```

Comment:
This needs to be ifdefed or we'll get compilation errors in puppet-agent builds. We might get it in AppVeyor too now that they're using the same GCC version.